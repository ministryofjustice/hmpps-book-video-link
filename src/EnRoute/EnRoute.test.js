import React from 'react'
import { shallow } from 'enzyme'

import EnRoute from './EnRoute'

describe('En route', () => {
  it('should render page correctly', () => {
    const movements = [
      {
        offenderNo: 'G0000GG',
        firstName: 'firstName',
        lastName: 'lastName',
        movementTime: '12:01:02',
        movementDate: '2019-10-10',
        fromAgencyDescription: 'Hull (HMP)',
        movementReasonDescription: 'Normal transfer',
        alerts: ['HA', 'XSA', 'XA', 'PEEP', 'XEL', 'XRF', 'XTACT'],
      },
    ]
    const wrapper = shallow(<EnRoute sortOrder="ASC" setColumnSort={jest.fn()} rows={movements} />)

    expect(wrapper).toMatchSnapshot()
  })
})
