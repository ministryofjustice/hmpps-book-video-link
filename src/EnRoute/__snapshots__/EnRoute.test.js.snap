// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`En route should render page correctly 1`] = `
<Fragment>
  <GridRow>
    <GridCol
      columnFull={false}
      columnOneHalf={false}
      columnOneQuarter={false}
      columnOneThird={false}
      columnThreeQuarters={false}
      columnTwoThirds={false}
      setWidth="one-quarter"
    >
      <SortLov
        setColumnSort={[MockFunction]}
        sortColumn="lastName"
        sortColumns={
          Array [
            "lastName",
          ]
        }
        sortOrder="ASC"
      />
    </GridCol>
  </GridRow>
  <table>
    <thead>
      <tr>
        <th
          className="straight width10"
        />
        <th
          className="straight width15"
        >
          <SortableColumn
            column="lastName"
            heading="Name"
            setColumnSort={[MockFunction]}
            sortColumn="lastName"
            sortOrder="ASC"
          />
        </th>
        <th
          className="straight width10"
        >
          Prison no.
        </th>
        <th
          className="straight width10"
        >
          D.O.B.
        </th>
        <th
          className="straight width10"
        >
          Departed
        </th>
        <th
          className="straight width15"
        >
          En route from
        </th>
        <th
          className="straight width15"
        >
          Reason
        </th>
        <th
          className="straight width25"
        >
          Flags
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        className="row-gutters"
        key="G0000GG_Hull (HMP)___"
      >
        <td
          className="row-gutters"
        >
          <OffenderImage
            offenderNo="G0000GG"
          />
        </td>
        <td
          className="row-gutters"
        >
          <OffenderName
            firstName="firstName"
            lastName="lastName"
          />
        </td>
        <td
          className="row-gutters"
        >
          G0000GG
        </td>
        <td
          className="row-gutters"
        >
          <FormattedDate />
        </td>
        <td>
          <div>
            <HoursAndMinutes
              hhmmss="12:01:02"
            />
          </div>
          <FormattedDate
            isoDate="2019-10-10"
          />
        </td>
        <td>
          Hull (HMP)
        </td>
        <td>
          Normal transfer
        </td>
        <td
          className="row-gutters"
        >
          <Flags
            alerts={
              Array [
                "HA",
                "XSA",
                "XA",
                "PEEP",
                "XEL",
                "XRF",
                "XTACT",
              ]
            }
            category=""
          />
        </td>
      </tr>
    </tbody>
  </table>
</Fragment>
`;
